core/src/epicli/cli/engine/BackupRecoveryEngineBase.py

```diff
@@ -0,0 +1,120 @@
+        self.build_directory = input_data.build_directory
+        self.manifest_docs = list()
+        self.input_docs = list()
+        self.configuration_docs = list()
+        self.cluster_model = None
+        self.backup_doc = None
+        self.recovery_doc = None
+        self.ansible_command = AnsibleCommand()
+
+    def __enter__(self):
+        super().__enter__()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        super().__exit__(exc_type, exc_value, traceback)
+
+    def _process_input_docs(self):
+        """Load, validate and merge (with defaults) input yaml documents."""
+
+        path_to_manifest = os.path.join(self.build_directory, MANIFEST_FILE_NAME)
+        if not os.path.isfile(path_to_manifest):
+            raise Exception('No manifest.yml inside the build folder')
+
+        # Get existing manifest config documents
+        self.manifest_docs = load_yamls_file(path_to_manifest)
+        self.cluster_model = select_single(self.manifest_docs, lambda x: x.kind == 'epiphany-cluster')
+
+        # Load backup / recovery configuration documents
+        self.input_docs = load_yamls_file(self.file)
+
+        # Validate input documents
+        with SchemaValidator(self.cluster_model, self.input_docs) as schema_validator:
+            schema_validator.run_for_individual_documents()
+
+        # Merge the input docs with defaults
+        with DefaultMerger(self.input_docs) as doc_merger:
+            self.input_docs = doc_merger.run()
+
+    def _process_configuration_docs(self):
+        """Populate input yaml documents with additional required ad-hoc data."""
+
+        # Seed the self.configuration_docs
+        self.configuration_docs = copy.deepcopy(self.input_docs)
+
+        # Please notice using DefaultMerger is not needed here, since it is done already at this point.
+        # We just check if documents are missing and insert default ones without the unneeded merge operation.
+        for kind in {'configuration/backup', 'configuration/recovery'}:
+            try:
+                # Check if the required document is in user inputs
+                document = select_single(self.configuration_docs, lambda x: x.kind == kind)
+            except ExpectedSingleResultException:
+                # If there is no document provided by the user, then fallback to defaults
+                document = load_yaml_obj(data_types.DEFAULT, 'common', kind)
+                # Inject the required "version" attribute
+                document['version'] = VERSION
+                # Copy the "provider" value from the cluster model
+                document['provider'] = self.cluster_model.provider
+                # Save the document for later use
+                self.configuration_docs.append(document)
+            finally:
+                # Copy the "provider" value to the specification as well
+                document.specification['provider'] = document['provider']
```

Comment:
Why we left so many comments in ready to use code?

Reply:
I think comparing to other pojects we don't have enough comments 🤔. Which ones do you want me to remove?

Reply:
No one. This is only generally question. I don't see many places commented like this. I suspect it's a new quality.